### ISSUE TITLE
Settings button: missing tooltip

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#7510]
The "Settings" button in the top bar used to have a tooltip displayed on hover and focus. This is how it looked like in Gutenberg 2.5:

<img width="399" alt="screen shot 2018-06-23 at 16 24 45" src="https://user-images.githubusercontent.com/1682452/41810691-fbdd631e-7702-11e8-888f-a8bb33d3a287.png">

On current master, no tooltip is displayed at all:

<img width="376" alt="no tooltip" src="https://user-images.githubusercontent.com/1682452/41810696-0c83e7d8-7703-11e8-9c8d-a9c3af233e47.png">

Aside:
the hover style is different too, not sure if that's intentional; if so, it should go in a separate issue though.

**Comment by afercia** [comment]
Looks like the `DotTip` added for the NUX prevents the tooltip from working correctly. As soon as I remove it, the tooltip is back. I think this ie because IconButtons are supposed to not use tooltips when they have children, assuming the child is some visible text in the button?

**Comment by afercia** [comment]
One option could be forcing the tooltip to appear, passing both the `label` and `tooltip` props:
```
label={ __( 'Settings' ) }
tooltip={ __( 'Settings' ) }
```

However, when the DotTips render, there's also a JS warning:
`Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.`

to my understanding, although the "tips" use a popover component, the "Settings" tip contains  a couple buttons and it's "hooked" within the Settings IconButton so apparently validateDOMNesting evaluates the nesting statically?

Regardless, moving the DotTip outside of the IconButton and wrapping both in a wrapper element could solve both issues.

**Comment by designsimply** [comment]
Tested and confirmed that hovering over the post settings gear icon does not result in showing a tooltip.

![screen shot 2018-06-25 at mon jun 25 4 52 06 pm](https://user-images.githubusercontent.com/1119271/41879827-7fdf385e-7898-11e8-9ba5-154beb102a95.png)
<sup>Seen at http://alittletestblog.com/wp-admin/post-new.php running WordPress 4.9.6 and Gutenberg 3.1.0 using Firefox 60.0.2 on macOS 10.13.5.</sup>

**Comment by afercia** [comment]
Just realized also the tooltip of the side inserter "Add block" button doesn't appear on hover any longer, for the same reason. To reproduce:
- create a new post
- hover on the "Write your story" area and then on the "plus" button

<img width="933" alt="screen shot 2018-07-21 at 19 54 12" src="https://user-images.githubusercontent.com/1682452/43038796-8eee4488-8d20-11e8-9797-df8248c3d308.png">

Actually, the `DotTip` passed as child to the Inserter ends up being a child of the IconButton rendered by Inserter. The IconButton component doesn't render a tooltip when it has children (because it assumes the children are visible button content), unless the tooltip is forced with the `tooltip` prop.

Worth reminding forcing the `tooltip` prop wouldn't solve the `validateDOMNesting` warning.

I'd suggest to give this issue higher priority, as the implementation of the NUX tips actually broke the tooltips behavior.
